### ISSUE TITLE
Possible False Positive CVE-2011-0638

### ISSUE THREAD

**Issue by obas21** [CISecurity/OVALRepo#1687]
oval:org.mitre.oval:def:12566  "Microsoft Windows Human Interface Device (HID) driver is prone to security bypass vulnerability", was written for XP, Vista, 7, 8, 2003, 2008, and 2012, but the check is using oval:org.mitre.oval:def:7133 that includes all Windows OS's. As a result this check is applicable for Win 10 and Win Server 2016 with the HID Service running, even though they are not identified as vulnerable in the CVE. 

If the vulnerability is valid for all Windows OS's, then the CVE needs to be updated to include Win 10 and 2016 in the description. If Win 10 and 2016 are not vulnerable, the OVAL check needs to be updated to exclude these OS's.

**Comment by DavidRies** [comment]
Hi @obas21,

OVAL doesn't formally define "applicability", but generally speaking the metadata/affected section indicates the platforms that the definition was intended to cover by the author.

As you said, that section indicates that this definition was written to cover XP, Vista, 7, 8, 2003, 2008, and 2012 (and 2008 R2). 

I would recommend against running this definition against other platforms, because you are likely to get incorrect results.

-David Ries

**Comment by obas21** [comment]
I think the issue is the check was written at a time when oval:def:7133 only included XP - 2012, but now that Windows 10 and Server 2016 are included in that def they will also be evaluated.

**Comment by DavidRies** [comment]
Hi @obas21,

They will only be evaluated if you run them on Windows 10 or Server 2016, and because this definition (CVE-2011-0638) isn't written for Windows 10 or Server 2016, you may not get an accurate result. This isn't really a content issue. It's a usage issue.

You can actually run any definition on any platform. You could run this definition on Ubuntu. But you may not get an accurate result. 

I'm not trying to be difficult or pedantic... it's a question of how the content is designed to be used and in this case, it's a convention and not a formal mechanism.

The convention is that platform-targeted definitions like the CVE-2011-0638 definition should only be run on the platforms that the author designed them for as indicated by the metadata/affected section.

If you run this definition on another platform, you may get an inaccurate result.

Does that make sense?

-David

**Comment by obas21** [comment]
When I look at the definition it is checking that "Microsoft Windows is installed". My understanding was that the defiinition should spell out each OS unless it trully is a vulnerability across all OS's. 
`		  <criteria>
		    <extend_definition comment="Microsoft Windows is installed" definition_ref="oval:org.mitre.oval:def:7133" />
		    <criterion comment="Check if HidServ service is running" test_ref="oval:org.mitre.oval:tst:100270" />
		  </criteria>`

**I believe it should be something like this, where it specifies each OS:**
`		      <criteria comment="Win 7 / R2 / Vista / 2K8" operator="OR">
		        <extend_definition comment="Microsoft Windows 7 (32-bit) is installed" definition_ref="oval:org.mitre.oval:def:6165" />
		        <extend_definition comment="Microsoft Windows 7 x64 Edition is installed" definition_ref="oval:org.mitre.oval:def:5950" />
		        <extend_definition comment="Microsoft Windows Server 2008 R2 x64 Edition is installed" definition_ref="oval:org.mitre.oval:def:6438" />
		        <extend_definition comment="Microsoft Windows Vista (32-bit) is installed" definition_ref="oval:org.mitre.oval:def:1282" />
		        <extend_definition comment="Microsoft Windows Vista x64 Edition is installed" definition_ref="oval:org.mitre.oval:def:2041" />
		        <extend_definition comment="Microsoft Windows Server 2008 (32-bit) is installed" definition_ref="oval:org.mitre.oval:def:4870" />
		        <extend_definition comment="Microsoft Windows Server 2008 (64-bit) is installed" definition_ref="oval:org.mitre.oval:def:5356" />`

Thanks,
Dave

**Comment by DavidRies** [comment]
@obas21, I agree that would make it a better definition and encourage you to contribute that improvement! -David
